doc/user_guide.md

```diff
@@ -0,0 +1,1518 @@
+The [`$project`](https://docs.mongodb.com/manual/reference/operator/aggregation/project/) stage
+removes them:  
+
+```json
+{
+  "$project": {
+    "cutoutScience": 0,
+    "cutoutTemplate": 0,
+    "cutoutDifference": 0
+  }
+}
+```
+
+3. Using the [`$lookup`](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/) stage, 
+the alert data are joined with auxiliary data stored in the `ZTF_alerts_aux` collection, which uses the alert's
+`objectId`s as unique document identifiers (`_id` - a concept in MongoDB similar to primary keys in SQL):
+
+```json
+  {
+    "$lookup": {
+      "from": "ZTF_alerts_aux",
+      "localField": "objectId",
+      "foreignField": "_id",
+      "as": "aux"
+    }
+  }
+```
+
+4. The final [`$project`](https://docs.mongodb.com/manual/reference/operator/aggregation/project/) stage reshapes 
+the joined data for convenience, selects the last 100 days of photometry history (which is done having practical
+considerations in mind and may be relaxed in the future), 
+and applies ACLs to the detection history stored in `prv_candidates`:
+
+```json
+{
+  "$project": {
+    "cross_matches": {
+      "$arrayElemAt": [
+        "$aux.cross_matches",
+        0
+      ]
+    },
+    "prv_candidates": {
+      "$filter": {
+        "input": {
+          "$arrayElemAt": [
+            "$aux.prv_candidates",
+            0
+          ]
+        },
+        "as": "item",
+        "cond": {
+          "$and": [
+            {
+              "$in": [
+                "$$item.programid",
+                [
+                  1,
+                  2,
+                  3
+                ]
+              ]
+            },
+            {
+              "$gt": [
+                {
+                  "$subtract": [
+                    "$candidate.jd",
+                    "$$item.jd"
+                  ]
+                },
+                100
+              ]
+            }
+          ]
+        }
+      }
+    },
+    "schemavsn": 1,
+    "publisher": 1,
+    "objectId": 1,
+    "candid": 1,
+    "candidate": 1,
+    "classifications": 1,
+    "coordinates": 1
+  }
+}
+```
+
+The user-defined filter stages then operate on the "enhanced" packets that look like this:
+
+![img/filter-04.png](img/filter-04.png) 
+
+`Fritz` automatically prepends these stages to all user-defined filters. However, when constructing/debugging
+filters in Compass, the users must take care of that -- all the examples below come with the upstream stages prepended.
+
+
+#### Limitations
+
+`$lookup` and `$unionWith` stages are not allowed in the user-defined part of the filters.
+
+### Filter examples
+
+Let us explore some examples! The provided JSON files contain prepended upstream stages, but they are not shown in the
+text below for brevity.
+
+#### Simple filter
+
+Let us start with a very simplistic example:
+
+[fritz_filter_101.json](data/filter_examples/fritz_filter_101.json)
```

Comment:
This is the right idea for a quick start, but I think we need something a bit meatier, that illustrates more of the functionality